src/pages/Bridge/IssueForm/index.tsx

```diff
@@ -263,21 +294,39 @@ const IssueForm = (): JSX.Element | null => {
       }
     };
 
+    const handleSelectVaultCheckboxChange = () => {
+      if (!isSelectVaultCheckboxDisabled) {
+        setSelectVaultManually(currentState => !currentState);
+      }
+    };
+
     const onSubmit = async (data: IssueFormData) => {
       try {
         setSubmitStatus(STATUSES.PENDING);
         await requestLimitsRefetch();
         await trigger(BTC_AMOUNT);
 
         const wrappedTokenAmount = BitcoinAmount.from.BTC(data[BTC_AMOUNT] || '0');
-
         const vaults = await window.bridge.vaults.getVaultsWithIssuableTokens();
-        const vaultId = getRandomVaultIdWithCapacity(Array.from(vaults), wrappedTokenAmount);
+
+        let vaultId: InterbtcPrimitivesVaultId;
+        let collateralTokenIdLiteral: CurrencyIdLiteral;
+
+        if (selectVaultManually) {
+          if (!vault) {
+            throw new Error('Specific vault is not selected!');
+          }
+          vaultId = vault[0];
+          collateralTokenIdLiteral = vault[0].currencies.collateral.asToken.toString();
+        } else {
+          vaultId = getRandomVaultIdWithCapacity(Array.from(vaults), wrappedTokenAmount);
+          collateralTokenIdLiteral = COLLATERAL_TOKEN_ID_LITERAL;
+        }
```

Comment:
Using react-hook-form's validation, we can avoid mixing these kinds of conditions with form submission by organizing in a more declarative way.

Reply:
Are you referring to L316 condition? Because other conditions aren't there to validate the form, but to handle issuing with specific and random vaults.

Reply:
@peterslany 
Oh, you're right. I'm mistaken. This is different from what I explained.